std/typecons.d

```diff
@@ -2068,35 +2068,55 @@ if (is(T == class) || is(T == interface) || isDynamicArray!T || isAssociativeArr
 Convenience function for creating a $(D Rebindable) using automatic type
 inference.
 
+If a $(LREF Rebindable) obj is passed to `rebindable`, it is returned without modifications.
+
 Params:
     obj = A reference to an object, interface, associative array, or an array slice
           to initialize the `Rebindable` with.
 
 Returns:
     A newly constructed `Rebindable` initialized with the given reference.
 */
-Rebindable!T rebindable(T)(T obj)
-if (is(T == class) || is(T == interface) || isDynamicArray!T || isAssociativeArray!T)
+auto rebindable(T)(T obj)
+if (is(T == class) || is(T == interface) || isDynamicArray!T || isAssociativeArray!T ||
+    __traits(isSame, TemplateOf!T, Rebindable))
 {
-    typeof(return) ret;
-    ret = obj;
-    return ret;
+    static if (is(T == class) || is(T == interface) || isDynamicArray!T || isAssociativeArray!T)
+    {
+        Rebindable!T ret;
+        ret = obj;
+        return ret;
+    }
+    else static if (__traits(isSame, TemplateOf!T, Rebindable))
+    {
+        return obj;
+    }
+    else
+    {
+        static assert(0, "Unknown type used.");
+    }
```

Comment:
Don't mix changes in functionality in with a documentation PR. Please split this off into a separate one.

Reply:
Hmm this is needed to be able to avoid duplicating the unittest for `rebindable` is has two different overloads and I thought that's trivial enough, but I can reverted it for now then.